libioc/Jail.py

```diff
@@ -589,50 +613,35 @@ def _stop_failed_jail(
                 )
             self._save_autoconfig()
 
-        if single_command is None:
-            # Start and Poststart Hooks
-            yield from self.__run_hook(
-                "start",
-                force=False,
-                event_scope=jailStartEvent.scope,
-                env=env
-            )
-            yield from self.__run_hook(
-                "poststart",
-                force=False,
-                event_scope=jailStartEvent.scope,
-                env=env
-            )
-        else:
-            yield from self.__run_hook(
-                "command",
-                force=False,
-                event_scope=jailStartEvent.scope,
-                script=str(single_command),
-                passthru=passthru,
-                env=env
-            )
-            if self.config["vnet"] is False:
-                yield from self._stop_non_vimage_network(
+        try:
+            if single_command is None:
+                # Start and Poststart Hooks
+                yield from self.__run_hook(
+                    "start",
                     force=False,
-                    event_scope=jailStartEvent.scope
+                    event_scope=jailStartEvent.scope,
+                    env=env
                 )
-            yield from self.__destroy_jail(event_scope=jailStartEvent.scope)
-            if (jid is not None) and (self.config["vnet"] is True):
-                yield from self.__stop_vimage_network(
-                    jid,
-                    event_scope=jailStartEvent.scope
+                yield from self.__run_hook(
+                    "poststart",
+                    force=False,
+                    event_scope=jailStartEvent.scope,
+                    env=env
                 )
-            yield from self.fstab.unmount(event_scope=jailStartEvent.scope)
-            yield from self.storage_backend.teardown(
-                self.storage,
-                event_scope=jailStartEvent.scope
-            )
-            yield from self.__clear_resource_limits(
-                force=False,
-                event_scope=jailStartEvent.scope
-            )
+            else:
+                yield from self.__run_hook(
+                    "command",
+                    force=False,
+                    event_scope=jailStartEvent.scope,
+                    script=str(single_command),
+                    passthru=passthru,
+                    env=env
+                )
+        except Exception as e:
+            yield jailStartEvent.fail(e)
+            raise e
 
+        _stop_jails()
```

Comment:
we don't catch the events from this function?

Reply:
Thanks for spotting the issue. It was patched with the latest commits.